cdc/log.cc

```diff
@@ -834,6 +834,8 @@ class transformer final {
      */
     std::unordered_set<const column_definition*> _non_atomic_column_deletes;
 
+    stats::part_type_set _touched_parts;
```

Comment:
I'm not sure why this change is needed. Maybe I will understand it while reading the following patches but just reading this patch it does not seem like a good idea. The previous approach was cleaner in my opinion.